src/views/Submit.vue

```diff
@@ -266,11 +289,31 @@ export default {
 			this.$refs.submitButton.click()
 		},
 
+		/*
+		 * Methods for catching unwanted unload events
+		 */
+		beforeWindowUnload(e) {
+			if (this.confirmLeaveForm()) {
```

Comment:
See above, the logic seems inverted, if leaving is confirmed we should *not* cancel the event.
```suggestion
			if (!this.confirmLeaveForm()) {
```